applications/trilinos_application/custom_strategies/builder_and_solvers/trilinos_block_builder_and_solver.h

```diff
@@ -567,6 +567,25 @@ class TrilinosBlockBuilderAndSolver
         if (BaseType::mDofSet.size()==0)
             KRATOS_ERROR << "No degrees of freedom!";
 
+        // If reactions are to be calculated, we check if all the dofs have reactions defined
+        // This is tobe done only in debug mode
+
+    #ifdef KRATOS_DEBUG        
+
+    if(BaseType::GetCalculateReactionsFlag())
+    {
+        int num_dofs =  static_cast<int> (BaseType::mDofSet.size());
+        bool all_dofs_have_reactions = true;
+  
+        for(int i = 0; i<num_dofs; i++)
+        { 
+            typename DofsArrayType::iterator dof_iterator = BaseType::mDofSet.begin() + i;
+            all_dofs_have_reactions = dof_iterator->HasReaction() && all_dofs_have_reactions;
```

Comment:
rather than doing this, you could throw the error as soon as you find one that is false

Reply:
And maybe print the id of the offending node and the name of the offending variable to help the user?

Reply:
Yes @RiccardoRossi  I finished it. I am now running tests. I will push it once I finish the tests.